genic/main.c

```diff
@@ -88,15 +89,35 @@ int main(int argc, char **argv)
   }
 
   /*First compute and write CDM*/
+  double mass[6] = {0};
+  /*Can neglect neutrinos since this only matters for the glass force.*/
+  compute_mass(mass, TotNumPart, TotNumPartGas, 0, 0);
   /*Not used*/
   int size[3], offset[3];
   int NumPart = get_size_offset(size, offset, All2.Ngrid);
```

Comment:
Currently NumPart in GenIC and NumPart in Gadget have different meanings. You may want to replace NumPart\> with NumPartCDM globally to avoid confusion.